File: app/src/main/java/com/pluu/webtoon/ui/MainActivity.java
package com.pluu.webtoon.ui;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.ActionBar;
import android.view.View;

import com.pluu.event.RxBusProvider;
import com.pluu.support.impl.ServiceConst.NAV_ITEM;
import com.pluu.webtoon.R;
import com.pluu.webtoon.common.Const;
import com.pluu.webtoon.common.PrefConfig;
import com.pluu.webtoon.event.ThemeEvent;
import com.pluu.webtoon.ui.settting.SettingsActivity;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.functions.Consumer;

public class MainActivity extends BaseNavActivity {

    private NAV_ITEM selfDrawerItem;

    @BindView(R.id.navTitle)
    View navTitle;

    private CompositeDisposable mCompositeDisposable;

    @Override
    protected NAV_ITEM getSelfNavDrawerItem() {
        return selfDrawerItem;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeButtonEnabled(true);
        }

        selfDrawerItem = PrefConfig.getDefaultWebToon(this);
        themeChange(new ThemeEvent(ContextCompat.getColor(this, selfDrawerItem.getColor()),
            ContextCompat.getColor(this, selfDrawerItem.getBgColor())));

        getSupportFragmentManager()
            .beginTransaction()
            .replace(R.id.container, MainFragment.newInstance(selfDrawerItem), Const.MAIN_FRAG_TAG)
            .commit();
    }

    void setSelfDrawerItem(NAV_ITEM selfDrawerItem) {
        this.selfDrawerItem = selfDrawerItem;
    }

    @Override
    protected void onResume() {
        super.onResume();
        mCompositeDisposable = new CompositeDisposable();
        mCompositeDisposable.add(
            RxBusProvider.getInstance()
                .toObservable()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(getBusEvent())
        );
    }

    @Override
    protected void onPause() {
        super.onPause();
        mCompositeDisposable.dispose();
    }

    @NonNull
    private Consumer<Object> getBusEvent() {
        return o -> {
            if (o instanceof ThemeEvent) {
                themeChange((ThemeEvent) o);
            }
        };
    }

    private void themeChange(ThemeEvent event) {
        navTitle.setBackgroundColor(event.getDarkColor());
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (resultCode == RESULT_OK) {
            Fragment fragment = getSupportFragmentManager().findFragmentByTag(Const.MAIN_FRAG_TAG);
            if (fragment != null) {
                fragment.onActivityResult(requestCode, resultCode, data);
            }
        }
    }

    @OnClick(R.id.btnSetting)
    void clickSetting() {
        Intent intent = new Intent(this, SettingsActivity.class);
        startActivity(intent);
        closeNavDrawer();
    }

}

File: app/src/main/java/com/pluu/webtoon/ui/MainActivity.kt
package com.pluu.webtoon.ui

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v4.content.ContextCompat
import com.pluu.event.RxBusProvider
import com.pluu.support.impl.NAV_ITEM
import com.pluu.webtoon.R
import com.pluu.webtoon.common.Const
import com.pluu.webtoon.common.PrefConfig
import com.pluu.webtoon.event.ThemeEvent
import com.pluu.webtoon.ui.settting.SettingsActivity
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.functions.Consumer
import kotlinx.android.synthetic.main.navdrawer.*

/**
 * 메인 화면 Activity
 * Created by pluu on 2017-05-07.
 */
class MainActivity : BaseNavActivity(), MainFragment.BindServiceListener {

    private var mCompositeDisposable = CompositeDisposable()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            setHomeButtonEnabled(true)
        }

        selfNavDrawerItem = PrefConfig.getDefaultWebToon(this)

        themeChange(ThemeEvent(ContextCompat.getColor(this, selfNavDrawerItem.color),
                ContextCompat.getColor(this, selfNavDrawerItem.bgColor)))

        supportFragmentManager
                .beginTransaction()
                .replace(R.id.container, MainFragment.newInstance(selfNavDrawerItem), Const.MAIN_FRAG_TAG)
                .commit()

        btnSetting.setOnClickListener {
            startActivity(Intent(this, SettingsActivity::class.java))
            closeNavDrawer()
        }
    }

    override fun onResume() {
        super.onResume()
        mCompositeDisposable.add(
                RxBusProvider.getInstance()
                        .toObservable()
                        .observeOn(AndroidSchedulers.mainThread())
                        .subscribe(busEvent)
        )
    }

    override fun onPause() {
        super.onPause()
        mCompositeDisposable.clear()
    }

    private val busEvent = Consumer<Any> {
        when (it) {
            is ThemeEvent -> themeChange(it)
        }
    }

    private fun themeChange(event: ThemeEvent) {
        navTitle.setBackgroundColor(event.darkColor)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (resultCode == Activity.RESULT_OK) {
            supportFragmentManager.findFragmentByTag(Const.MAIN_FRAG_TAG)?.
                    onActivityResult(requestCode, resultCode, data)
        }
    }

    override fun bindNavItem(item: NAV_ITEM) {
        selfNavDrawerItem = item
    }
}
